guides/bookshop-astro-guide/using-components.mdx

```diff
@@ -0,0 +1,130 @@
+---
+_schema: guide-item
+_uuid: 11fcf7d3-e787-437f-a542-2915f6c5e056
+_created_at: "2023-02-01 02:40:30 +0000"
+title: "Using Bookshop components"
+nav_title: "Using Bookshop components"
+order: 4
+published: true
+image: /documentation/static/bookshop-11ty/bookshop-astro-og.png
+description: "Using your Astro Bookshop components on your site"
+tags: []
+related_articles: null
+related_links: []
+---
+
+Using a Bookshop component on your Astro site is similar to using a regular component with a few differences.
+
+## The Bookshop component directive
+
+The Bookshop integration provides a `bookshop:live` for use in your Astro template files.
+
+If you've been following along with this guide, add the following snippet
+to a page to use the sample component we created:
+
+<comp.CodeBlock language="astro" source="src/pages/index.astro">
+```
+---
+import Sample from '../components/sample/sample.astro'
+---
+<Sample bookshop:live text="Hello from the sample component" />
+```
+</comp.CodeBlock>
+
+This matches the standard syntax for using a component in an Astro file but with the added `bookshop:live` directive.
+Bookshop processes components with this directive at build time to set up live visual editing and visual data bindings.
+
+The `bookshop:live` directive can only be used in Astro templates, but there are no limitations on where those templates can be located.
+For example, you could enable live editing for a component only within a certain layout or parent component. You can
+also add the `bookshop:live` directive to React components imported in your Astro templates.
+
+Like other Astro directives, the `bookshop:live` directive must be visible at compile time. For example, the following code would not work:
```

Comment:
Not sure that giving a code block that explicitly will _not_ work is ideal. Maybe something like:

"For example, the following code would require <specific code> at <specific line number> in order for the directive to be visible when compiling."

Alternatively, give fully working code and point out or comment in the specific directive that makes it work.